server/src/main/java/org/elasticsearch/index/mapper/BlockDocValuesReader.java

```diff
@@ -652,12 +652,12 @@ private BlockLoader.Block readSingleDoc(BlockFactory factory, int docId) throws
         }
 
         @Override
-        public BlockLoader.Block read(BlockFactory factory, Docs docs) throws IOException {
+        public BlockLoader.Block read(BlockFactory factory, Docs docs, int offset) throws IOException {
             if (docs.count() == 1) {
                 return readSingleDoc(factory, docs.get(0));
             }
             try (BlockLoader.SingletonOrdinalsBuilder builder = factory.singletonOrdinalsBuilder(ordinals, docs.count())) {
-                for (int i = 0; i < docs.count(); i++) {
+                for (int i = offset; i < docs.count(); i++) {
```

Comment:
I think `factory.singletonOrdinalsBuilder(ordinals, docs.count()))` (the previous line) can produce incorrect results if the offset is not considered.

Reply:
++. got it. will add tests that catch this.